enterprise/views.py

```diff
@@ -232,13 +255,15 @@ def post_course_specific_consent(self, request, course_id, consent_provided):
         if not request.user.is_authenticated():
             raise Http404
 
-        EnterpriseCourseEnrollment.objects.update_or_create(
-            enterprise_customer_user__user_id=request.user.id,
-            course_id=course_id,
-            defaults={
-                'consent_granted': consent_provided,
-            }
-        )
+        enrollment_deferred = request.POST.get('enrollment_deferred')
+        if enrollment_deferred is None:
+            EnterpriseCourseEnrollment.objects.update_or_create(
+                enterprise_customer_user__user_id=request.user.id,
+                course_id=course_id,
```

Comment:
Should we perform `course_id` validation in `EnterpriseCourseEnrollmentWriteSerializer` as well?